src/Adyen/HttpClient/CurlClient.php

```diff
@@ -51,6 +51,11 @@ public function requestJson(\Adyen\Service $service, $requestUrl, $params)
 			'User-Agent: ' . $userAgent
 		);
 
+		// if idempotency key is provided as option include into request
+        if ($requestOptions != null && !empty($requestOptions['idempotencyKey'])) {
```

Comment:
```suggestion
        if (!empty($requestOptions['idempotencyKey'])) {
```

Checking if null is not necessary - empty will take care of that